src/playground/playground.js

```diff
@@ -40,6 +40,57 @@ const getAssetUrl = function (asset) {
     return assetUrlParts.join('');
 };
 
+const addProfilerPanels = function (
+    stats, vm, panelName = 'BLK%', panelFg = '#fff', panelBg = '#111') {
+
+    vm.runtime.enableProfiling();
+    if (vm.runtime.profiler === null) {
+        // Profiler isn't available on the local system.
+        return;
+    }
+
+    const blockPercentPanel = stats.addPanel(new window.Stats.Panel(panelName, panelFg, panelBg));
+
+    const profiler = vm.runtime.profiler;
+    vm.runtime.profiler = null;
```

Comment:
Why are you setting this to null here?

Reply:
This line sets profiler to null to disable profiling until we want to record since active profiling slows down the scratch-vm a little.

The runtime's profiler member is null by default. To ensure errors don't occur the instrumented areas check if profiler is not null before calling `profiler.start` as that would throw a TypeError because you cannot read any properties of null.

So toggling the profiler on, is setting profiler to a Profiler instance. And toggling off, is setting it to null.

Reply:
I guess another to mention is this line is assuming that profiling is initially off.